influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -193,74 +195,103 @@ impl<W: Wal, T: TimeProvider> WriteBufferImpl<W, T> {
         &self,
         database_name: &str,
         table_name: &str,
-        _filters: &[Expr],
-        _projection: Option<&Vec<usize>>,
-        _ctx: &SessionState,
+        filters: &[Expr],
+        projection: Option<&Vec<usize>>,
+        ctx: &SessionState,
     ) -> Result<Vec<Arc<dyn QueryChunk>>, DataFusionError> {
         let db_schema = self
             .catalog
             .db_schema(database_name)
             .ok_or_else(|| DataFusionError::Execution(format!("db {} not found", database_name)))?;
-        let table = db_schema
-            .tables
-            .get(table_name)
-            .ok_or_else(|| DataFusionError::Execution(format!("table {} not found", table_name)))?;
-        let schema = table.schema.clone();
 
-        let table_buffers = self.clone_table_buffers(database_name, table_name);
-        let chunks = table_buffers
-            .into_iter()
-            .map(|table_buffer| {
-                let batch = table_buffer.rows_to_record_batch(&schema, table.columns());
-                let batch_stats = create_chunk_statistics(
-                    Some(table_buffer.row_count()),
-                    &schema,
-                    Some(table_buffer.timestamp_min_max()),
-                    None,
-                );
-
-                let chunk: Arc<dyn QueryChunk> = Arc::new(BufferChunk {
-                    batches: vec![batch],
-                    schema: schema.clone(),
-                    stats: Arc::new(batch_stats),
-                    partition_id: TransitionPartitionId::new(
-                        TableId::new(0),
-                        &table_buffer.segment_key,
-                    ),
-                    sort_key: None,
-                    id: ChunkId::new(),
-                    chunk_order: ChunkOrder::new(0),
-                });
-
-                chunk
-            })
-            .collect();
+        let table_schema = {
+            let table = db_schema.tables.get(table_name).ok_or_else(|| {
+                DataFusionError::Execution(format!(
+                    "table {} not found in db {}",
+                    table_name, database_name
+                ))
+            })?;
 
-        Ok(chunks)
-    }
+            table.schema.clone()
+        };
+
+        let segment_state = self.segment_state.read();
+        let mut chunks =
+            segment_state.get_table_chunks(db_schema, table_name, filters, projection, ctx)?;
+        let parquet_files = segment_state.get_parquet_files(database_name, table_name);
+
+        let mut chunk_order = chunks.len() as i64;
+        let object_store_url = self.persister.object_store_url();
+
+        for parquet_file in parquet_files {
+            // TODO: update persisted segments to serialize their key to use here
+            let partition_key = data_types::PartitionKey::from(parquet_file.path.clone());
```

Comment:
I just needed a string to uniquely identify the partition key. Edge doesn't really have a concept of partitions. The closest to it would be segments, so ideally what we'd use would be the segment key (which is the string that is the time that looks like an RFC3339). Right now we only have one parquet file per segment, so the file path works as a unique key.

Ideally, we'd use the segment key, but the PersistedSegment needs to be updated with the data for that. I've logged #24755 to track.